filemutex_flock.go

```diff
@@ -57,3 +58,15 @@ func (m *FileMutex) RUnlock() {
 	}
 	m.mu.RUnlock()
 }
+
+// Close does an Unlock() combined with closing and unlinking the associated
+// lock file. You should create a New() FileMutex for every Lock() attempt if
+// using Close().
+func (m *FileMutex) Close() {
+	if err := syscall.Flock(m.fd, syscall.LOCK_UN); err != nil {
+		panic(err)
+	}
+	syscall.Close(m.fd)
+	syscall.Unlink(m.path)
+	m.mu.Unlock()
```

Comment:
As for preventing Unlock() before Close()... again, I can only think the solution is explicit documentation, and allow the panic to happen. If sync.Mutex Unlock() -> Unlock() can panic, so should FileMutex.Unlock() -> Close(); they're equally wrong.